### PR TITLE
Refactor IAccount type to only include path and index for mnemonics

### PR DESCRIPTION
**Pull request by Mrtenz**

### CODE REVIEW

**Comment by Mrtenz** [comment]
[ch8561]

File: src/app/screens/SignTransaction/SignTransaction.tsx [hunk 1]
```diff
@@ -36,9 +36,22 @@ export const SignTransaction = () => {
   }
 
   if (currentAccount) {
+    if (currentAccount.type === WalletType.MNEMONIC) {
```
Comment (FrederikBolding) [inline src/app/screens/SignTransaction/SignTransaction.tsx:39]: Why is this necessary? This was much cleaner without this.

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: src/app/screens/SignTransaction/SignTransaction.tsx (hunk 1 above)
Comment (Mrtenz) [reply src/app/screens/SignTransaction/SignTransaction.tsx:39]: To properly infer the type of `currentAccount`. Otherwise, the type would be inferred as `never`. Alternatively I can do a type cast if you think that's better?

Comment (FrederikBolding) [reply src/app/screens/SignTransaction/SignTransaction.tsx:39]: I would prefer that over this yeah.

**Review by FrederikBolding — APPROVED** [review]
